### PR TITLE
Fix issues with startup in Windows

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
Don't load the kernelparams grains on Windows
Properly detect the hostname

### What issues does this PR fix or reference?
Found in testing

### Previous Behavior
The following lines were showing up in the minion start up:
```
[ERROR   ] Having trouble getting a hostname.  Does this machine have its hostname and domain set properly?
[DEBUG   ] Failed to read /proc/cmdline: [Errno 2] No such file or directory: u'/proc/cmdline'
```

### Tests written?
No

### Commits signed with GPG?
Yes

### CODE REVIEW

File: salt/utils/network.py [hunk 1]
```diff
@@ -208,7 +208,7 @@ def get_fqhostname():
         )
         for info in addrinfo:
             # info struct [family, socktype, proto, canonname, sockaddr]
-            if len(info) >= 4:
+            if len(info) >= 4 and info[3]:
```
Comment (rallytime) [inline salt/utils/network.py]: I'm not quite sure what this check is for. Can you explain this a little more?

Comment (dwoz) [reply salt/utils/network.py]: `canonname` can be an empty string. We don't want to populate the `l` list with empty names. @twangboy , maybe add a comment about this?

Comment (twangboy) [reply salt/utils/network.py]: It's making sure there's a value in info[3]. It's sometimes `''` which causes the function to return None on Windows.

Comment (rallytime) [reply salt/utils/network.py]: OK, yes - let's get a comment in here so it doesn't get removed in the future.

**Review by rallytime — APPROVED** [review]

**Pull request merged**